EasyPost.Tests/BatchTest.cs

```diff
@@ -40,154 +43,159 @@ private static async Task<Batch> CreateOneCallBuyBatch()
 
-            Assert.IsTrue(batches.Count <= Fixture.PageSize);
-            Assert.IsNotNull(batchCollection.has_more);
-            foreach (var item in batches)
+            Assert.True(batches.Count <= Fixture.PageSize);
+            foreach (Batch item in batches)
             {
-                Assert.IsInstanceOfType(item, typeof(Batch));
+                Assert.IsType<Batch>(item);
             }
         }
 
-        [TestMethod]
-        public async Task TestCreateAndBuy()
+        [Fact]
+        [CrudOperations.Update]
+        public async Task TestAddRemoveShipment()
         {
-            _vcr.SetUpTest("create_and_buy");
+            UseVCR("add_remove_shipment");
+
+            Shipment shipment = await Client.Shipment.Create(Fixture.OneCallBuyShipment);
+
+            Batch batch = await Client.Batch.Create();
 
-            Batch batch = await Batch.CreateAndBuy(new Dictionary<string, object>
+            batch = await batch.AddShipments(new List<Shipment>
             {
-                {
-                    "shipments", new List<Dictionary<string, object>>
-                    {
-                        Fixture.OneCallBuyShipment
-                    }
-                }
+                shipment
             });
 
-            Assert.IsInstanceOfType(batch, typeof(Batch));
-            Assert.IsTrue(batch.id.StartsWith("batch_"));
-            Assert.AreEqual(1, batch.num_shipments);
+            Assert.Equal(1, batch.num_shipments);
+
+            batch = await batch.RemoveShipments(new List<Shipment>
+            {
+                shipment
+            });
+
+            Assert.Equal(0, batch.num_shipments);
         }
 
-        [TestMethod]
+        [Fact]
+        [CrudOperations.Update]
         public async Task TestBuy()
         {
-            _vcr.SetUpTest("buy");
+            UseVCR("buy");
 
             Batch batch = await CreateOneCallBuyBatch();
 
-            await batch.Buy();
+            batch = await batch.Buy();
 
-            Assert.IsInstanceOfType(batch, typeof(Batch));
-            Assert.AreEqual(1, batch.num_shipments);
+            Assert.IsType<Batch>(batch);
+            Assert.Equal(1, batch.num_shipments);
         }
 
-        [TestMethod]
+        [Fact]
+        [CrudOperations.Update]
         public async Task TestCreateScanForm()
         {
-            _vcr.SetUpTest("create_scan_form");
+            UseVCR("create_scan_form");
 
 
             Batch batch = await CreateOneCallBuyBatch();
 
-            await batch.Buy();
-
-            if (_vcr.IsRecording())
+            if (IsRecording())
             {
                 Thread.Sleep(10000); // Wait enough time to process
             }
 
-            await batch.GenerateScanForm();
+            batch = await batch.Buy();
+
+            if (IsRecording())
+            {
+                Thread.Sleep(30000); // Wait enough time to process
+            }
+
+            batch = await batch.GenerateScanForm();
```

Comment:
Not in my experience lately.

Reply:
If for whatever reason someone needs to come bump this because our queue is backed up, they can make the one-off tweak to a higher time. We have 6 other libraries that have never needed longer than 10 seconds over various re-recordings. Having the test suite wait an entire minute (for the two 30 second waits added here) when it may not always be necessary is too long and will only frustrate users (aka: us). Can we please lower it to 10 seconds?

Reply:
EDIT: After looking, it seems you reduced these to 10s. Please also mention what you did when responding to feedback. It greatly reduces the back and forth instead of requiring a reviewer to dig into each change individually.